stan/math/prim/scal/fun/fdim.hpp

```diff
@@ -23,11 +20,8 @@ namespace math {
  */
 template <typename T1, typename T2>
 inline return_type_t<T1, T2> fdim(T1 x, T2 y) {
-  typedef return_type_t<T1, T2> return_t;
-  using std::numeric_limits;
-  if (is_any_nan(x, y))
-    return NOT_A_NUMBER;
-  return (x <= y) ? 0 : x - y;
+  using std::fdim;
+  return fdim(x, y);
```

Comment:
Since there is a fdim method in var/fvar I think we can do the same templating trick here as well

Reply:
I'm pretty sure we can use auto as well since this works on scalars. @bob-carpenter is return_type<T> something from pre-C++11 or will auto deduction not work for these for some reason?